### ISSUE TITLE
LSP Message: File is not being analyzed

### ISSUE THREAD

**Issue by qhoirulanwar** [Dart-Code/Dart-Code#4654]
Request textDocument/hover failed.
  Message: File is not being analyzed
  Code: -32007 

Flutter 3.13.0-6.0.pre.
Dart SDK version: 3.1.0-321.0.dev
Vscode 1.80.1
Dart  v3.68.1
Windows 10 Home 22H2

**Comment by DanTup** [comment]
@qhoirulanwar can you provide some more context for this error? Are you just seeing it in the VS Code developer console, or is it also causing issues?

It's known that this error can happen at startup (where VS Code sends requests for the server has started analyzing) but it shouldn't cause any problems beyond that.

If you're having problems, there might be another issue too.

**Comment by qhoirulanwar** [comment]
intellisense not show, and i see this error
I have tried to restart the analysis server, but that didn't work

**Comment by DanTup** [comment]
Oh thanks, I see you're on Flutter master. I think you're hitting https://github.com/dart-lang/sdk/issues/53000 which was a bug introduced last week. A fix landed in Dart yesterday, but it needs to roll into Flutter (but I think is clocked for other reasons).

I would suggest switching to the Flutter stable or beta branch for now until this fix has landed in Flutter master (or, you could revert to a slightly older version of Flutter master from around a week ago).

**Comment by qhoirulanwar** [comment]
oh ok, currently I'm using dart intellisense vscode plugin version **3.22.0** which runs normally

**Comment by DanTup** [comment]
That's a very old version of the extension, you might find many things don't work on the latest SDKs (for example it will try to run the `pub` executable that was removed).

The reason that version works is that it uses the legacy protocol. You could also force that on the new versions by adding the following to your VS Code user settings:

```
"dart.useLegacyAnalyzerProtocol": true
```

There may still be things that won't work when using the legacy protocol, but it would be much less than using a 2year old version of the extension. I would recommend removing this setting in a week or so when the fix has landed in the Flutter though - use of the legacy protocol in the extension is not being maintained.

**Comment by DanTup** [comment]
FWIW the Dart->Flutter roll was unblocked and the fix has now landed in Flutter master, so you can use the latest VS Code extension again and latest Flutter master without hitting this issue.
